### DISCUSSION TITLE
returning the Query type from a mutation

### DISCUSSION THREAD

**Discussion by martysaxton** [MichalLytek/type-graphql#779] [Q&A]
Looking for a way to achieve this.  

From:  https://graphql-rules.com/rules/mutation-payload-query

```graphql
type Mutation {
  likePost(id: 1): LikePostPayload
}

type LikePostPayload {
   record: Post
   query: Query
}
```

Also discussed here:  https://medium.com/@stubailo/returning-the-query-type-in-graphql-111d5c0b15b8

**Comment by MichalLytek** [comment]
Described in #427 as an anti-pattern and won't be supported.

**Comment by mmahalwy** [comment]
Adding to this. If not a query, but is returning a payload with the resources supported or must be a separate type? (see: https://github.com/MichalLytek/type-graphql/issues/780#issuecomment-773103558)

**Reply by MichalLytek** [reply]
You can create a generic type like `Response<T>` with fields `errors` and `data` (or something you need), so you can easily reduce the boilerplate

**Reply by mmahalwy** [reply]
yeah, i guess that could work. i don't know if the key can be dynamic, no? so `data` could be whatever passed?

**Reply by MichalLytek** [reply]
Dynamic you mean TS "any"? Then you can use GraphQLJSONScalar but then you can't select subfields in graphql query.

With the generic type you can do
```ts
const UserResponse = Response(User);
const AnimalResponse = Response(Animal);
```
etc.
And then use those types in your queries/mutations
